### PR TITLE
test(insights): Run all tests with `filters`-based insights using HogQL

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Let's make the HogQL engine the default. Still leaving a strategy with the legacy engine running.
This will allow us to catch problems in the saved insights/dashboards layer, and later on when converting `/insight/{trend,funnel,etc}`.

### CODE REVIEW

File: posthog/api/insight.py [hunk 1]
```diff
@@ -326,7 +326,6 @@ def create(self, validated_data: dict, *args: Any, **kwargs: Any) -> Insight:
                     raise serializers.ValidationError("Dashboard not found")
 
                 DashboardTile.objects.create(insight=insight, dashboard=dashboard, last_refresh=now())
-                insight.last_refresh = now()  # set last refresh if the insight is on at least one dashboard
```
Comment (Twixes) [inline posthog/api/insight.py:329]: This has not been used in a long time and not sure it ever actually worked. The actual `last_refresh` is in InsightCachingState at the insight level, and in the cached result packages themselves

File: posthog/caching/calculate_results.py [hunk 2]
```diff
@@ -145,6 +146,12 @@ def calculate_for_query_based_insight(
     if isinstance(response, BaseModel):
         response = response.model_dump()
 
+    update_cached_state(  # Updating the relevant InsightCachingState
+        insight.team_id,
+        response.get("cache_key"),
+        response.get("last_refresh"),
+        result=None,  # Not caching the result here, since in HogQL this is the query runner's responsibility
+    )
```
Comment (Twixes) [inline posthog/caching/calculate_results.py:154]: The only behavioral change in this PR: `TestDashboard::test_refresh_cache` revealed we weren't updating `InsightCachingState` in here. This change brings us fully back on track with tracking which insights need a background refresh

File: posthog/caching/calculate_results.py [hunk 3]
```diff
@@ -145,13 +146,23 @@ def calculate_for_query_based_insight(
...
 
+    cache_key = response.get("cache_key")
+    last_refresh = response.get("last_refresh")
+    if isinstance(cache_key, str) and isinstance(last_refresh, str):
+        update_cached_state(  # Updating the relevant InsightCachingState
+            insight.team_id,
+            cache_key,
+            last_refresh,
+            result=None,  # Not caching the result here, since in HogQL this is the query runner's responsibility
+        )
```
Comment (Twixes) [inline posthog/caching/calculate_results.py:157]: The only behavioral change in this PR: `TestDashboard::test_refresh_cache` revealed we weren't updating `InsightCachingState` in here. This change brings us fully back on track with tracking which insights need a background refresh

**Review by thmsobrmlr — APPROVED** [review]
Changes look good. 👍 to merge this in once tests pass

> This will allow us to catch problems in the saved insights/dashboards layer, and later on when converting /insight/{trend,funnel,etc}.

I don't think we ever want to convert these, since they are replaced by `/query`, right?

**Comment by Twixes** [comment]
As for `/insight/{trend,funnel,etc}`, the issue is that we definitely want to throw out the code of legacy insights, and these run-`filters`-ad-hoc endpoints are a blocker in their current state. There are two paths that I see right now:

1. We just remove them – but it's certain that there are users somewhere with cron jobs or whatever programmed to get data out of PostHog by hitting these endpoints. Pulling the plug would cause those users massive pain.
2. We keep the endpoints, but wrap them all in `filters_to_query()`. The legacy code gets the boot, while API users are (generally) unharmed.

What do you think @thmsobrmlr? Let me know if you have an idea to refine this

**Pull request merged**